ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -197,6 +202,22 @@ boolean isOperationComplete() {
   }
 
   /**
+   * Notify for overall blob creation if the operation is complete and the blob was put successfully. Also ensure that
+   * notifications have been sent out for all successfully put data chunks.
+   */
+  void maybeNotifyForBlobCreation() {
+    if (isOperationComplete()) {
+      metadataPutChunk.maybeNotifyForFirstChunkCreation();
```

Comment:
Refactored this a little bit, hopefully it's a bit cleaner now. I can't call `getNumDataChunks()` if chunk filling did not complete successfully, so it's not quite this simple.